src/System.Memory/src/System/ReadOnlySpan.Portable.cs

```diff
@@ -193,6 +193,10 @@ public override string ToString()
         {
             if (typeof(T) == typeof(char))
             {
+                if (_pinnable is string str)
```

Comment:
This is not enough. You have to also check that the offset and length are for the whole string (ie that the Span is not pointing at substring).

Reply:
Would adding offset and length checks be sufficient here?

```C#
if (obj is string str && _byteOffset == 0 && _length == str.Length)
```

Reply:
You need to use `StringAdjustment` for the offset.